### PR TITLE
Support to labels endpoint - filter by labelname

### PR DESCRIPTION
**Pull request by shinusuresh**
PR for #255

### CODE REVIEW

**Comment by cdancy** [comment]
@shinusuresh looks like a failing mock test?

**Comment by shinusuresh** [comment]
My bad @cdancy. Missed to add mock JSON file.

**Comment by shinusuresh** [comment]
@cdancy Can you have a look into this PR, please

File: src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java [hunk 1]
```diff
@@ -45,4 +47,12 @@
...
     @GET
     LabelsPage list(@Nullable @QueryParam("prefix") String prefix);
+
+    @Named("labels:get-by-name")
+    @Documentation({"https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-rest.html#idp88"})
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/api/{jclouds.api-version}/labels/{labelName}")
+    @Fallback(BitbucketFallbacks.LabelsByNameOnError.class)
+    @GET
+    Labels getLabelByName(@Nullable @PathParam("labelName") String labelName);
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java:57]: @shinusuresh should we allow `@Nullable` here? What happens if a user passes in null?

**Comment by cdancy** [comment]
@shinusuresh one comment to address. Are there any labels we could query for that would be present by default for integration tests?

File: src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java (hunk 1 above)
Comment (shinusuresh) [reply src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java:57]: @cdancy Its, not mandatory as per the spec. 
I think it needs to be handled. I am updating PR

**Comment by shinusuresh** [comment]
> @shinusuresh one comment to address. Are there any labels we could query for that would be present by default for integration tests?
It's a little tricky. Labels are being set at the project level, and these API's are not tied to any project. 
I think when we implement https://docs.atlassian.com/bitbucket-server/rest/6.0.0/bitbucket-rest.html#idp227, these test can be amended to create a label first and then assert.
What you think?

File: src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java (hunk 1 above)
Comment (shinusuresh) [reply src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java:57]: @cdancy Is it a `NullPointerException` expected if the user passes in null?

Comment (cdancy) [reply src/main/java/com/cdancy/bitbucket/rest/features/LabelsApi.java:57]: Can we just remove the @Nullable annotation and force users to pass in the parameter? Feels odd to have it here.

Also ... can we rename the `Labels` object to just `Label` as that would better represent what is being returned. Pluralizing it gives the impression that potentially multiple labels are getting returned.

**Comment by cdancy** [comment]
> amended to create a label first and then assert.
> What you think @cdancy ?

That works for me.

**Review by cdancy — APPROVED** [review]
LGTM

**Comment by cdancy** [comment]
@shinusuresh assuming no last minute issues all looks good to me and thanks for addressing the comments. I can merge once CI testing has passed. Did you want a new release with these changes or did you want to wait a bit and submit some more code first? I'm fine either way and will leave it up to you.

**Comment by shinusuresh** [comment]
> Did you want a new release with these changes

@cdancy thanks for checking. Not needed now. Will release it, once Labels API is finished.

**Pull request merged**